### ISSUE TITLE
[Feature Request] Support 3.26 topbar transparency switching

### ISSUE THREAD

**Issue by CharoSW** [passingthru67/workspaces-to-dock#117]
By this I mean dynamically change the transparency of the dock when any window is maximised like how the topbar now works.  Currently the dock looks a little out of place in one of the two situations depending on your selected transparency level

**Comment by passingthru67** [comment]
I agree that dynamic transparency will have to be supported to fit in with Gnome 3.26. In fact, I've been watching the progress of this feature in dash-to-dock and wandering when it would finally be merged into the master branch. Well, that happened this week. I plan on adding this feature also, as soon as I get a chance.

**Comment by passingthru67** [comment]
Here's a question for anybody interested in this topic. How should workspaces-to-dock transparency work in relation to the top panel and dash-to-dock?

There are 3 scenarios I can think of:

1) If dash-to-dock is not present (or disabled) -> Sync with the top panel. But, should we control the top panel transparency like dash-to-dock adaptive mode (see paragraph after scenario #3 below) or offer that as an option?

2) If dash-to-dock is enabled but not using adaptive opacity -> Should we sync with the top panel or dash-to-dock or offer an option for the user to select?

3) If dash-to-dock is enabled and using adaptive opacity -> Sync with dash-to-dock since it controls the transparency of the top panel also.

Currently, dash-to-dock hijacks control of the top panel transparency when the 'Adaptive' option is selected under Appearance - Customize opacity. Basicly, if there are windows near the dash or near the panel, the transparency is adjusted on both.

**Comment by CharoSW** [comment]
For 1. offer the option with it enabled by default would probably be the best. That way you get consistency while also giving people the option to disable should they wish

For 2. sync with top panel would probably be the more more expected behaviour

**Comment by konnjuta** [comment]
Highly anticipating this feature. Hoping it will be similar to dash-to-dock 'Adaptive' option.
